core/src/builder/transaction/txhandler.rs

```diff
@@ -211,43 +270,6 @@ impl TxHandler<Unsigned> {
         Ok(())
     }
 
-    // Candidate refactoring
```

Comment:
Do you need to keep this code? Are you really going to use it in the future?

Reply:
I was thinking of doing something like this instead of script indexing (i.e. using magic numbers). We didn't do that. I don't really think this is an optimal solution (as it requires looping over all scripts), so decided to just remove it.

Also don't understand the comment here, do you want it back? I removed the lines (did not add them)

Reply:
Since we're now using SpendPaths, we'd need to embed some kind of script selector closure (like `Box<dyn FnMut(&dyn SpendableScript) -> bool>`) into the SpendPath. That's going to be kind of expensive and slow